src/browser/core/db.ts

```diff
@@ -0,0 +1,263 @@
+    }[]
+  > {
+    return this.getDB(database).getTableKeys(table, schema);
+  }
+
+  createCancellableQuery(database: string, queryId: string, queryText: string): Promise<void> {
+    this.cancellableQueries[queryId] = this.getDB(database).query(queryText);
+    return Promise.resolve();
+  }
+
+  async cancelCancellableQuery(queryId: string): Promise<void> {
+    const query = this.cancellableQueries[queryId];
+    try {
+      if (query) {
+        await query.cancel();
+      }
+    } finally {
+      delete this.cancellableQueries[queryId];
+    }
+  }
+
+  async executeCancellableQuery(queryId: string): Promise<QueryRowResult[]> {
+    try {
+      const query = this.cancellableQueries[queryId];
+      if (query) {
+        const results = await query.execute();
+        return results;
+      }
+
+      return [];
+    } finally {
+      delete this.cancellableQueries[queryId];
+    }
+  }
+
+  executeQuery(database: string, queryText: string): Promise<QueryRowResult[]> {
+    return this.getDB(database).executeQuery(queryText);
+  }
+
+  getQuerySelectTop(database: string, table: string, schema?: string): Promise<string> {
+    return this.getDB(database).getQuerySelectTop(table, schema);
+  }
+
+  getTableCreateScript(database: string, table: string, schema?: string): Promise<string[]> {
+    return this.getDB(database).getTableCreateScript(table, schema);
+  }
+
+  getTableSelectScript(database: string, table: string, schema?: string): Promise<string> {
+    return this.getDB(database).getTableSelectScript(table, schema);
+  }
+
+  getTableInsertScript(database: string, table: string, schema?: string): Promise<string> {
+    return this.getDB(database).getTableInsertScript(table, schema);
+  }
+
+  getTableUpdateScript(database: string, table: string, schema?: string): Promise<string> {
+    return this.getDB(database).getTableUpdateScript(table, schema);
+  }
+
+  getTableDeleteScript(database: string, table: string, schema?: string): Promise<string> {
+    return this.getDB(database).getTableDeleteScript(table, schema);
+  }
+
+  getViewCreateScript(database: string, table: string, schema?: string): Promise<string[]> {
+    return this.getDB(database).getViewCreateScript(table, schema);
+  }
+
+  getRoutineCreateScript(
+    database: string,
+    table: string,
+    type: string,
+    schema?: string,
+  ): Promise<string[]> {
+    return this.getDB(database).getRoutineCreateScript(table, type, schema);
+  }
+
+  truncateAllTables(database: string, schema?: string): Promise<void> {
+    return this.getDB(database).truncateAllTables(schema);
+  }
+
+  getTableColumnNames(database: string, table: string, schema?: string): Promise<string[]> {
+    return this.getDB(database).getTableColumnNames(table, schema);
+  }
+
+  setSelectLimit(limit: number): void {
+    return setSelectLimit(limit);
+  }
+}
+
+// Keep connections by BrowserWindow instances
+const connections: {
+  [windowId: string]: SqlectronDB;
+} = {};
+
```

Comment:
Because now the db connection lives in the browser process, it means when we there are multiple application windows open they share the same browser process, and to allow each window to have its own db connection, they are stored by windowId in this object.